### DISCUSSION TITLE
How can templates for WinUI project creation be better?

### DISCUSSION THREAD

**Discussion by Rinku-Stephen** [microsoft/microsoft-ui-xaml#9921] [General]
Currently we have few variations of blank templates. We want to improve on this experience and understand what else we can introduce. Let us know your preferences, early thoughts or suggestions here.

**Comment by myokeeh** [comment]
My goto is Template Studio. I know that's probably not what you meant, but I really don't use the templates in VS. I've used Template Studio to create apps.

**Reply by ghost1372** [reply]
It is very complicated and heavy

**Comment by mdtauk** [comment]
There should be some templates that include the customised modern Titlebar, as well as the System Backdrop.

In fact, the Windows Design Team should codify the basic "silhouettes" of what modern WinUI apps should look like, and include the bare bones code and comments using best practices to ensure adhering to these designs are easy to do.

But it becomes a bit more complex when we factor in considering templates for Win32 / WinForms apps, including WinUI content as islands.  Should there be item templates for inserting Xaml Islands or Continents into WinForms and WPF?  I think only the Windows devs can answer that, as they have the experience of inserting Xaml content into legacy apps, which is not super common yet outside of Microsoft's internal teams.

**Comment by MPITech** [comment]
I love WinUICommunity's method of setting up a new app.  It is how I started mine and it has worked out so well for us.  It would be really beneficial for all Visual Studio users to have something like this to get started, but with better explanations of each option.

https://github.com/WinUICommunity/WinUICommunity

![1 1](https://github.com/user-attachments/assets/d984074a-00d3-453d-86a1-4b73758e3d00)

**Comment by JimmyRespawn** [comment]
A better store packaging template please. 

Now store bundle[msixupload] packaging is very confusing. In UWP, I can create a store bundle within 1 minute. Now I have to generate each platform's bundle -> Make appx bundle by command -> make failed -> figure out why. After couple hours searching. No anwser. Roll back to UWP.

**Reply by ghost1372** [reply]
Fortunately, they are fixing the bug
https://github.com/microsoft/microsoft-ui-xaml/discussions/9807

![image](https://github.com/user-attachments/assets/6ab7c697-00fd-4d6f-bbf0-351d7c26684e)

**Comment by saint4eva** [comment]
Ability to boostrap a winui app from the commandline and developing and running using visual studio code. E.g. dotnet new winui --mode packaged. After which I can run it using dotnet run or dotnet publish --arch ARM64 or X64

**Comment by sterenas** [comment]
We’d love to have VB.NET templates for WinUI projects.  Would be the way to go to modernize our apps. Pleaseeeeee

**Reply by MPITech** [reply]
I'm sure you don't need me chiming in, but take it from a long time VB+WinForms developer... the move to C# and WinUI wasn't nearly as daunting as I thought it would be.  C# has some fantastic syntax features that VB does not.  My advice if you want to modernize your apps is to get a VB-to-C Visual Studio extension, convert your VB code to C#, and slowly integrate it into your new WinUI app, making changes as needed.  It's not that bad!  :)  Good luck.

**Comment by hippieZhou** [comment]
I hope to create a standard implementation, such as a store-style layout in the home and settings pages.

**Comment by ghost1372** [comment]
i use [WinUICommunity-Templates](https://github.com/winUICommunity/WinUICommunity-Templates), This template helps me create the following features quickly:
- Home Landing Page
- Settings Page
- Theme Settings Page
- Custom TitleBar
- NavigationView
- Add Custom Libraries
and more...

![image](https://github.com/user-attachments/assets/ed57ae54-74cf-49dd-93d3-07e4b4244205)

**Comment by Rinku-Stephen** [comment]
Hi Everyone - While we parse your comments and figure out what's the right next steps.. I would request you to check out the documentation here to understand how to build simple layouts and sihouettes - https://learn.microsoft.com/en-us/windows/apps/design/basics/app-silhouette .. We will try to include grid layouts to this..

**Comment by Mr-Racnok** [comment]
My published app won't launch , anyone ??

**Reply by JimmyRespawn** [reply]
For duo package template. Same for me. 
If you unzip the package generated by vs. You'll find out the core appx bundles are not packaged into msixupload bundle.

**Comment by mcNets** [comment]
WinUICommunity templates seems the way to go.

**Reply by hippieZhou** [reply]
Honestly, I don't like using `WinUICommunity`, because I think it's much more than I want.
